XUtils.c

```diff
@@ -12,6 +12,7 @@ in the source distribution for its full text.
 #include <assert.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <limits.h> // IWYU pragma: keep
```

Comment:
Not needed, when included in `XUtils.h` …

Reply:
The code below uses `CHAR_BIT`. This include line hints the direct dependency.

Reply:
If the public interface (`XUtils.h`) already depends on the same header, there's no practical need to repeat this in `XUtils.c`. The additional comment is just a marker for IWYU, preventing the header from being removed when those two conditional compilation blocks are skipped.

Reply:
What if IWYU warn me about the use of `CHAR_BIT` in XUtils.c without including `<limits.h>`? This is what IWYU is supposed to do. Did I get anything wrong?

Reply:
Okay, IMHO it should already detect the situation that the header is included in the accompanying header, but I'll check. Leave it in for now; doesn't hurt. ;-)